NuRadioMC/simulation/simulation.py

```diff
@@ -790,11 +733,9 @@ def run(self):
                             electric_field.set_trace_start_time(trace_start_time)
                             electric_field[efp.azimuth] = azimuth
                             electric_field[efp.zenith] = zenith
-                            electric_field[efp.ray_path_type] = self._prop.solution_types[r.get_solution_type(iS)]
+                            electric_field[efp.ray_path_type] = self._prop.solution_types[self._raytracer.get_solution_type(iS)]
                             electric_field[efp.nu_vertex_distance] = sg['travel_distances'][iSh, channel_id, iS]
                             electric_field[efp.nu_viewing_angle] = viewing_angles[iS]
-                            electric_field[efp.reflection_coefficient_theta] = r_theta
```

Comment:
I just notice another thing. Can we save all ray tracing parameters to the efield? At the moment we don't know which efield corresponds to which ray tracing solution. 
We could just save the new "propagation output dict" as efield parameter.

Reply:
We can do that, but it's a separate issue, so let's do it in another pull request. This one is already complex enough